PZNS_Framework/media/lua/client/03_mod_core/PZNS_NPCGroup.lua

```diff
@@ -1,22 +1,112 @@
-PZNS_NPCGroup = {};
-PZNS_NPCGroup.__index = PZNS_NPCGroup;
-
---- Cows: can set the group leader later.
----@param groupID any
----@return table
-function PZNS_NPCGroup:newGroup(
-    groupID
+require("00_references/init")
+
+---@class Group
+---@field groupID groupID           Unique group identifier
+---@field name string               Group name
+---@field leaderID survivorID       NPC ID of group leader
+---@field members table<survivorID> list of group members
+---@field memberCount integer       number of members in group
+---@field factionID factionID?      Faction this group belongs to
+local Group = {}
```

Comment:
Please add a prefix "PZNS_", this will help make searching much easier in the future.
"Group" -> "PZNS_Group"

"Group" is too common and that will make searching difficult.